Mathlib/Tactic/NormNum/NormNumI.lean

```diff
@@ -0,0 +1,235 @@
+/-
+Copyright (c) 2025 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth, Yunzhou Xie, Sidharth Hariharan
+-/
+import Mathlib.Analysis.Normed.Ring.Basic
+import Mathlib.Data.Complex.Basic
+
+/-!
+## `norm_num` extension for complex numbers
+
+This file provides a `norm_num` extension for complex numbers, allowing the computation of
+additions, multiplications, inversions, conjugates, and powers of complex numbers.
+
+-/
+
+open Lean Meta Elab Qq Tactic Complex Mathlib.Tactic
+open ComplexConjugate
+
+namespace Mathlib.Meta
+namespace NormNumI
+
+theorem split_I : I = ⟨0, 1⟩ := rfl
```

Comment:
For symmetry with the internals of norm_num, I'd suggest
```suggestion
structure IsComplex (z : ℂ) (re im : ℝ) : Prop where
  out : z = ⟨re, im⟩

theorem split_I : IsComplex I 0 1 where out := rfl
```
(and throughout)


Some other benefits of this approach:
* Dot notation
* It's easier to do a future refactor to make it use `IsROrC` instead.